mlir/include/mlir/Dialect/ArmSME/Transforms/Passes.td

```diff
@@ -122,4 +122,23 @@ def TileAllocation
   let dependentDialects = ["func::FuncDialect"];
 }
 
+def VectorLegalization
+  : Pass<"arm-sme-vector-legalization", "mlir::ModuleOp"> {
+  let summary = "Legalize vectors for ArmSME";
+  let description = [{
+    This pass legalizes vector operations so that they can be lowered to ArmSME.
+    This includes decomposing operations that operate on vector types larger
+    than a single SME tile (e.g. `vector<[8]x[8]xf32>`) into multiple SME
```

Comment:
`vector<[8]x[8]xf32>` matches the size of `ZA`, right? How about vectors that are larger than this? Or smaller?

Reply:
Right - but are there any limitations? I know that there are -  those are effectively enforced by `isMultipleOfSMETileVectorType`, right? It would be good add a few more comments here, e.g.:
* "legal" in this context means something that matches SMEs virtual tiles as available in hardware
* "input" type must be a multiple of these tiles (so definitely 2d and scalable in both dimensions)

Reply:
> legal - means something that can be lowered to ArmSME 

Perhaps you could try to defining that? I assume it means either:
* 2d scalable vectors with sizes matching SME virtual tiles,
* 1d scalable vectors
* n-D vectors with at most the trailing dim scalable?